h2/src/main/org/h2/value/Transfer.java

```diff
@@ -530,6 +530,13 @@ public void writeValue(Value v) throws IOException {
             }
             break;
         default:
+            if (JdbcUtils.customDataTypesHandler != null) {
+                DataType dt = JdbcUtils.customDataTypesHandler.getDataTypeById(type);
```

Comment:
Lets assume that type handler knows all the types and will throw an exception if not. Lets get rid of `getDataTypeById(type) == null` checks everywhere.